File: ClinicaManagement/ClinicaManagement/ClinicaModel/Persona.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClinicaModel
{
    public abstract class Persona
    {
        private static int contador = 0;
        private int idPersona;
        private string nombre;
        private string apellido;
        private string correo;
        private int telefono;
        private string direccion;
        private DateTime fechaNacimiento;
        private char genero;
        private string dni;

        public int IdPersona { get; set; }
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public string Correo { get; set; }
        public int Telefono { get; set; }
        public string Direccion { get; set; }
        public DateTime FechaNacimiento { get; set; }
        public char Genero { get; set; }
        public string DNI { get; set; }

        public Persona(){ }

        public Persona(string nombre, string apellido, string correo, int telefono, string direccion, DateTime fechaNacimiento, char genero, string dni)
        {
            IdPersona = ++contador;
            Nombre = nombre;
            Apellido = apellido;
            Correo = correo;
            Telefono = telefono;
            Direccion = direccion;
            FechaNacimiento = fechaNacimiento;
            Genero = genero;
            DNI = dni;
        }

    }
}

File: Proyecto Java/SoftCapsuleCare-Users/src/pe/edu/pucp/citamedica/model/usuario/Persona.java
package pe.edu.pucp.citamedica.model.usuario;

import java.text.SimpleDateFormat;
import java.util.Date;

public class Persona {
    
    private static int contador = 0;
    private int idPersona;
    private String nombre;
    private String apellido;
    private String correoElectronico;
    private int numTelefono;
    private String direccion;
    private Date fechaNacimiento;
    private char genero; // M O F
    private String DNI;

    public Persona() {
    }

    public Persona(String nombre, String apellido, String correoElectronico, int numTelefono, String direccion, 
                   Date fechaNacimiento, char genero, String DNI) {
        this.idPersona = ++contador;
        this.nombre = nombre;
        this.apellido = apellido;
        this.correoElectronico = correoElectronico;
        this.numTelefono = numTelefono;
        this.direccion = direccion;
        this.fechaNacimiento = fechaNacimiento;
        this.genero = genero;
        this.DNI = DNI;
    }
    
    public int getIdPersona() {
        return idPersona;
    }

    public void setIdPersona(int idPersona) {
        this.idPersona = idPersona;
    }
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public String getCorreoElectronico() {
        return correoElectronico;
    }

    public void setCorreoElectronico(String correoElectronico) {
        this.correoElectronico = correoElectronico;
    }

    public int getNumTelefono() {
        return numTelefono;
    }

    public void setNumTelefono(int numTelefono) {
        this.numTelefono = numTelefono;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public Date getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(Date fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public char getGenero() {
        return genero;
    }

    public void setGenero(char genero) {
        this.genero = genero;
    }

    public String getDNI() {
        return DNI;
    }

    public void setDNI(String DNI) {
        this.DNI = DNI;
    }

    // Declaramos el método como abstracto ya que Persona es abstracta
    public void actualizarInformacion(){
        
    }
    
}
